src/btdmp.cpp

```diff
@@ -21,35 +18,38 @@ void Btdmp::Reset() {
 }
 
 void Btdmp::Tick(u64 ticks) {
-    if (transmit_enable) {
-        transmit_timer += ticks;
-        if (transmit_timer >= transmit_period) {
-            transmit_timer = 0;
-            std::array<std::int16_t, 2> sample;
-            for (int i = 0; i < 2; ++i) {
-                if (transmit_queue.empty()) {
-                    std::printf("BTDMP: transmit buffer underrun\n");
-                    sample[i] = 0;
-                } else {
-                    sample[i] = static_cast<s16>(transmit_queue.front());
-                    transmit_queue.pop();
-                    transmit_empty = transmit_queue.empty();
-                    transmit_full = false;
-                    if (transmit_empty) {
-                        interrupt_handler();
-                    }
+    if (!transmit_enable) {
+        return;
+    }
+
+    transmit_timer += ticks;
```

Comment:
Not sure how many ticks you typically process at once, but `transmit_timer` being a `uint16_t` value and `transmit_period` being 4096 makes this dangerously close to overflowing.

Reply:
Typical tick count for Citra/Panda/<secret dog emulator> is 16K, don't remember what MelonDS uses but something similar I think

Reply:
https://github.com/melonDS-emu/melonDS/blob/ea1755bed0378a2f3783e78c5e63c266583218ce/src/DSi_DSP.cpp#L569

Ditto I guess

Reply:
Should probably bump this to a u32 to be safe yeah. Typical slice length is 16K as mentioned by peach. Block sizes range between 20 and 100 cycles max and core timing is ticked every time in the dispatcher. Though I want to refactor this to only check for interrupts/tick core timing when an interrupt is close to occuring and execute code with a fast dispatcher in between these